File: xiaoming/1004/LongestCommonPrefix.java

public class LongestCommonPrefix {
	
	public String longestCommonPrefixF(String[] strs) {
        String res = "";
        if (strs == null || strs.length == 0) return res;
        if (strs.length == 1) return strs[0];
        for (int i=0; i<strs[0].length(); i++) {
            for (int j = 1; j<strs.length; j++) {
                if (strs[j].length() == i) return res;  
                if (strs[0].charAt(i) != strs[j].charAt(i)) return res; 
            }
            res += strs[0].charAt(i);
        }
        return res;
    }
}

File: mu_wang/10_01/Longest Common Prefix.py
Longest Common Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        prefix = ""
        for i in zip(*strs):
            bog = set(i)
            if len(bog)==1:
                prefix+=bog.pop()
            else:
                break
        return prefix
        